src/Cassandra/Requests/BatchRequest.cs

```diff
@@ -97,6 +90,26 @@ internal class BatchRequest : ICqlRequest
             return timestamp != long.MinValue ? (long?) timestamp : null;
         }
 
+        /// <summary>
+        /// Gets the serial consistency level of the request or null if not defined.
+        /// </summary>
+        /// <exception cref="NotSupportedException" />
+        private static ConsistencyLevel? GetSerialConsistencyLevel(IStatement statement, QueryOptions queryOptions)
+        {
+            var consistency = queryOptions.GetSerialConsistencyLevel();
+            if (statement.SerialConsistencyLevel != ConsistencyLevel.Any)
+            {
+                consistency = statement.SerialConsistencyLevel;
+            }
```

Comment:
💯 about code duplication on `QueryProtocolOptions`.

I'm really eager to remove `QueryProtocolOptions` but the problem is that is part of the public API since 1.0.

Anyway, I've followed your advice on code reuse and introduced a method: `GetSerialConsistencyLevelOrDefault()`.